openhands/integrations/gitlab/gitlab_service.py

```diff
@@ -523,6 +527,196 @@ async def create_mr(
+        # Extract name without extension
+        name = file_name.replace('.md', '').replace('.cursorrules', 'cursorrules')
+
+        return {
+            'name': name,
+            'path': path,
+            'created_at': datetime.now().isoformat(),
+        }
+
+    async def _process_cursorrules_file(
+        self, client: httpx.AsyncClient, base_url: str
+    ) -> dict | None:
+        """Check if .cursorrules file exists and return basic metadata."""
+        try:
+            headers = await self._get_gitlab_headers()
+            cursorrules_response = await client.get(
+                f'{base_url}/repository/files/.cursorrules/raw', headers=headers
+            )
+            if cursorrules_response.status_code == 200:
+                return self._create_microagent_response('.cursorrules', '.cursorrules')
+        except Exception as e:
+            logger.warning(f'Error checking .cursorrules: {str(e)}')
+        return None
+
+    async def _process_gitlab_md_files(
+        self, client: httpx.AsyncClient, tree_items: list
+    ) -> list[dict]:
+        """Process .md files from GitLab tree items without fetching content."""
+        microagents = []
+
+        for item in tree_items:
+            if (
+                item['type'] == 'blob'
+                and item['name'].endswith('.md')
+                and item['name'] != 'README.md'
+            ):
+                try:
+                    microagents.append(
+                        self._create_microagent_response(item['name'], item['path'])
+                    )
+                except Exception as e:
+                    logger.warning(
+                        f'Error processing microagent {item["name"]}: {str(e)}'
+                    )
+
+        return microagents
+
+    async def get_microagents(self, repository: str) -> list[dict]:
+        """Fetch microagents from GitLab repository using GitLab API.
+
+        Args:
+            repository: Repository name in format 'owner/repo' or 'domain/owner/repo'
+
+        Returns:
+            List of microagents found in the repository (without content for performance)
+        """
+        microagents_path = self._determine_microagents_path(repository)
+
+        # URL encode the repository name for GitLab API
+        project_id = repository.replace('/', '%2F')
+
+        # Determine base URL - check if it's a self-hosted GitLab instance
+        if '/' in repository and not repository.startswith('gitlab.com/'):
+            # Handle self-hosted GitLab (e.g., 'gitlab.example.com/owner/repo')
+            parts = repository.split('/')
+            if len(parts) >= 3:
+                domain = parts[0]
+                project_id = '/'.join(parts[1:]).replace('/', '%2F')
+                base_url = f'https://{domain}/api/v4/projects/{project_id}'
+            else:
+                base_url = f'{self.BASE_URL}/projects/{project_id}'
+        else:
+            base_url = f'{self.BASE_URL}/projects/{project_id}'
```

Comment:
self.BASE_URL is initialized based on the self hosted domain already ; so I believe we only need to extract `project_id` reliably here